### PR TITLE
e2e_node: use IPv4 for node name lookup

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:
The tests may fail because we resolve an IPv6 for a node: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-crio-cgroupv1-node-e2e-features/1673951855526285312

We now force IPv4 usage as well as use a context for lookup.

cc @dims @SergeyKanzhelev @harche @haircommander 

#### Which issue(s) this PR fixes:

Testgrid: https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv1-node-e2e-features

Follow-up on https://github.com/kubernetes/kubernetes/pull/118910

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: test/e2e_node/util.go [hunk 1]
```diff
@@ -640,20 +640,18 @@ func WaitForPodInitContainerToFail(ctx context.Context, c clientset.Interface, n
 	})
 }
 
-func nodeNameOrIP() string {
+func nodeNameOrIP(ctx context.Context) string {
 	// Check if the node name in test context can be resolved
-	if ips, err := net.LookupIP(framework.TestContext.NodeName); err != nil {
+	if ips, err := net.DefaultResolver.LookupIP(ctx, "ip4", framework.TestContext.NodeName); err != nil {
```
Comment (SergeyKanzhelev) [inline test/e2e_node/util.go:645]: another reason to use `localhost` instead. I don't think we will have problems resolving localhost. See thread: https://github.com/kubernetes/kubernetes/pull/118910#discussion_r1244247229

wdyt @saschagrunert ?

/lgtm
/approve
/hold

**Comment by SergeyKanzhelev** [comment]
/close

in favor of https://github.com/kubernetes/kubernetes/pull/118932/

File: test/e2e_node/util.go (hunk 1 above)
Comment (aojea) [reply test/e2e_node/util.go:645]: if you can, just use localhost instead of the instance name

**Pull request closed without merging**